Source/WebCore/loader/HistoryController.cpp

```diff
@@ -161,7 +163,7 @@ void HistoryController::restoreScrollPositionAndViewState()
         auto desiredScrollPosition = currentItem->shouldRestoreScrollPosition() ? currentItem->scrollPosition() : view->scrollPosition();
         LOG(Scrolling, "HistoryController::restoreScrollPositionAndViewState scrolling to %d,%d", desiredScrollPosition.x(), desiredScrollPosition.y());
         if (page && frame->isMainFrame() && currentItem->pageScaleFactor())
-            page->setPageScaleFactor(currentItem->pageScaleFactor() * page->viewScaleFactor(), desiredScrollPosition);
+            page->chrome().client().setPageScaleFactor(currentItem->pageScaleFactor(), desiredScrollPosition, [] { });
```

Comment:
I'm not sure about this.  It's pretty tangled up right now, but I think we want to move it to the UI process being in charge of setting this.  Probably in the future once the history in general is more UI process based.  Charlie's working on that.